### PR TITLE
[4.0] Fix Expected 'string' found 'int' error in com_content

### PR DESCRIPTION
**Pull request by bahl24**
Pull Request for Issue - Fixes Expected 'string' found 'int' error in com_content. .

### Summary of Changes
Changed to string

### Expected result
string type

### Actual result
int type

### CODE REVIEW

**Comment by SharkyKZ** [comment]
Judging by the PR that added button classes (#18501), value typehints were added by mistake and should be removed. @asika32764?

**Comment by Hackwar** [comment]
I too think that this is not the right approach and we should rather fix the method than to mask this like this.

**Comment by asika32764** [comment]
When I writing this method I think state value will be loaded from SQL so it will often be string.

But for real use, I agree it should remove `string` type hint.

**Comment by asika32764** [comment]
If change the `addState()` signature, we should also modify `getState()` 

https://github.com/joomla/joomla-cms/pull/18501/files#diff-72e4f228b49d04896cfdffaa98f94045R131

**Comment by asika32764** [comment]
We have workflow system now, maybe this class should also rewrite to fit workflow.

**Comment by bahl24** [comment]
@asika32764 Are you working on this or should I try to modify the `addState()` and other functions in a separate PR as there are many other functions with such type hints?

**Comment by asika32764** [comment]
Since I'm joining some serious projects I'm afraid of I will unable to continue work with this function. If someone want to modify this method, just try to pass the test, then I think it will work fine.

**Comment by bahl24** [comment]
Ok, let me try

File: libraries/src/Button/ActionButton.php [hunk 1]
```diff
@@ -95,7 +95,7 @@ protected function preprocess()
 	/**
 	 * Add a state profile.
 	 *
-	 * @param   string  $value    The value of this state.
+	 * @param   		$value    The value of this state.
```
Comment (Quy) [inline libraries/src/Button/ActionButton.php:98]: I assume type should always be provided.

**Comment by wilsonge** [comment]
No action here and I'm still yet to here of anyone else getting the original issue so I'm closing this for now. It can be reopened if we reproduce this in the future.

**Comment by SharkyKZ** [comment]
This doesn't cause a real issue now because we don't use `strict_types` directive. I think this should be changed anyways. We use only integer states (only passed as string if from DB). Do we want to support non-integer states here (for 3rd parties)?

**Comment by wilsonge** [comment]
We should typehint to an integer earlier i assume (and typehint method accordingly)

**Comment by SharkyKZ** [comment]
Looks like this won't work because internally `Joomla\CMS\Button\ActionButton` uses `_default` as some sort of default state.

https://github.com/joomla/joomla-cms/blob/9e20bcd4da1a7d54c8860a0f00cf7416abc4958c/libraries/src/Button/ActionButton.php#L173

**Comment by wilsonge** [comment]
Ugh and that is typehinted to string 🤦‍♂ . So I guess does someone want to do this but with correct doc blocks?

**Comment by SharkyKZ** [comment]
See PR https://github.com/joomla/joomla-cms/pull/28380.

**Pull request closed without merging**